File: workloads/python/mix_copy_popitem_first_and_last.py
"""
Creates a main dictionary with 'E' number of elements
Creates 'D' number of copies of the main dictionary
Pops each element in all dictionaries
Pops either first or last element by chance
"""
from collections import OrderedDict
import time
import random


E = 100 # Number of elements to create in dictionary
D = 1000 # Number of dictionaries to create and pop

start = time.clock()
main_dict = OrderedDict()

for i in range(E): #Populate main dictionary
    main_dict[i] = i

for j in range(D):
    temp_dict = main_dict.copy() #Create new copy of main dictionary

    for k in range(temp_dict.__len__()): #Pop all items in temp dictionary by chance
        rnd = random.randint(0, 1)
        if rnd == 0:
            temp_dict.popitem(last=False)
        else:
            temp_dict.popitem(last=True)

    temp_dict.clear()

main_dict.clear()

print(time.clock()-start)

File: python/common/org/python/workloads/mix_copy_popitem_first_and_last.java
package python.common.org.python.workloads;

import org.python.stdlib.collections.OrderedDict;
import org.python.types.Int;
import java.util.concurrent.ThreadLocalRandom;


import java.util.*;

public class mix_copy_popitem_first_and_last {
    public static void main(String args[]) {

        int E = 100; //Number of elements to create in dictionary
        int D = 1000; // Number of dictionaries to create and pop

        long start = System.currentTimeMillis();

        org.python.Object[] dictargs = {null};
        Map<String, org.python.Object> kwargs = new HashMap<String, org.python.Object>();
        OrderedDict main_dict = new OrderedDict(dictargs, kwargs);

        for(int i = 0; i < E; i++) { //Populate main dictionary
            main_dict.__setitem__(Int.getInt(i), Int.getInt(i));
        }

        for(int j = 0; j < D; j++) { //Create new copy of main dictionary
            OrderedDict temp_dict = (OrderedDict) main_dict.copy(); //Create new copy of main dictionary
            for(int k = 0; k < E; k++) { //Pop all items in temp dictionary by chance
                int rnd = ThreadLocalRandom.current().nextInt(0, 1 + 1);

                if (rnd == 0) {
                    temp_dict.popitem(org.python.types.Bool.FALSE);
                }
                else {
                    temp_dict.popitem(org.python.types.Bool.TRUE);
                }
            }
            temp_dict.clear();
        }
        main_dict.clear();
        System.out.println(System.currentTimeMillis()-start + "ms");
    }
}
